src/charm.py

```diff
@@ -109,6 +136,61 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         container.replan()
         self._update_app_and_unit_status(ActiveStatus())
 
+    def _database_uri(self) -> typing.Dict[str, str]:
```

Comment:
Should this be in a separate database module?

Reply:
Done